syntax = "proto3";

package steeleagle;

message ComputeRequest {
    bytes frame_data = 1;
    int32 frame_width = 2;
    int32 frame_height = 3;
}

message ComputeResult {
    repeated AIDetection compute_result = 1;
}

message AIDetection {
    int64 timestamp_ns = 1;
    int32 class_id = 2;
    int32 frame_id = 3;
    string class_name = 4;
    string cam = 5;
    float class_confidence = 6;
    float detection_confidence = 7;
    float x_min = 8;
    float y_min = 9;
    float x_max = 10;
    float y_max = 11;
}
